proposals/stdlib/char-int-conversions.md

```diff
@@ -0,0 +1,184 @@
+
+## Motivation
+
+The conversions above are often found confusing. People calling `Char.toInt()` usually expect to get the digit value of char, 
+similar to how `String.toInt()` works. Reverse conversions like `Double.toChar()` usually make no sense, 
+and it would be more clear if the intent was expressed explicitly by converting the number to `Int` first 
+and then getting the character corresponding to that `Int` code.
+
+Some examples of the confusion:
+
+* https://discuss.kotlinlang.org/t/convert-char-to-byte-gives-wrong-result/11548
+* https://stackoverflow.com/questions/47592167/how-do-i-convert-a-char-to-int
+* https://stackoverflow.com/questions/51961220/what-happens-with-toint
+* https://stackoverflow.com/questions/52393028/how-to-convert-character-to-its-integer-value-in-kotlin
+* https://stackoverflow.com/questions/57420203/how-to-convert-digit-to-character-in-kotlin
+* https://stackoverflow.com/questions/57515225/why-it-kotlin-giving-me-the-wrong-int-value-when-converting-from-a-string
+* https://stackoverflow.com/questions/61712411/converting-big-number-into-string-and-then-splitting-into-single-digits-results
+* https://blog.jdriven.com/2019/10/converting-char-to-int-in-kotlin/
+
+To alleviate the confusion, we would like to deprecate the conversion functions above 
+and introduce new ones with the names that make it clear what the function purpose is. 
+
+## Description
+
+For each `N` type where `N` is `Int`, `Short`, `Long`, `Byte`, `Double`, `Float`
+we are going to:
+
+* Deprecate `Char.toN()` functions
+* Deprecate `N.toChar()` functions
+  
+We also need to deprecate `Number.toChar()` function, change its modality from `abstract` to `open`, and provide 
+the default implementation of this function, `toInt().toChar()`.
+
+* Introduce functions to get the integer code of a `Char` and to construct a `Char` from the given code.
+
+```kotlin
+/**
+ * Creates a Char with the specified [code], or throws an exception if the [code] is out of `Char.MIN_VALUE.code..Char.MAX_VALUE.code`.
+ * 
+ * If the program that calls this function is written in a way that only valid [code] is passed as the argument,
+ * using the overload that takes a [UShort] argument is preferable (`Char(intValue.toUShort())`).
+ * That overload doesn't check validity of the argument, and may improve program performance when the function is called routinely inside a loop.
+ */
+fun Char(code: Int): Char
+
+/**
+ * Creates a Char with the specified [code].
+ */
+fun Char(code: UShort): Char
+
+/**
+ * Returns the code of this Char.
+ *
+ * Code of a Char is the value it was constructed with, and the UTF-16 code unit corresponding to this Char.
+ */
+val Char.code: Int
+```
+These functions will be proposed as replacements for the deprecated conversions above.
+For example: 
+```kotlin
+char.toInt() -> char.code
+char.toShort() -> char.code.toShort()
+
+int.toChar() -> Char(int.toUShort())
```

Comment:
Worth to mention that we plan to stabilize `UShort` as well, so such migration will be stable

Reply:
Done